### PR TITLE
Add support for Deno location config

### PR DESCRIPTION
**Pull request by joshgillies**
Fixes: #2730

### CODE REVIEW

File: compiler-core/src/config.rs [hunk 1]
```diff
@@ -713,6 +713,8 @@ pub struct DenoConfig {
     pub allow_all: bool,
     #[serde(default)]
     pub unstable: bool,
+    #[serde(default)]
+    pub location: String,
```
Comment (lpil) [inline compiler-core/src/config.rs:717]: How about we make this a `Uri`?

**Review by lpil — COMMENTED** [review]
Fab! Would you mind updating the changelog also? Thanks

**Comment by lpil** [comment]
I think you may not have pushed all the commits yet- it does not build successfully.

File: compiler-core/src/config.rs (hunk 1 above)
Comment (joshgillies) [reply compiler-core/src/config.rs:717]: Good call, I'll update to use the `Uri` type now 👍🏻

**Comment by joshgillies** [comment]
@lpil thanks for the feedback! Have just pushed some more changes, including a Deno test project. Will also update the changelog 👍🏻

File: compiler-core/src/config.rs [hunk 2]
```diff
@@ -818,6 +820,26 @@ mod uri_serde {
...
+        let string: Option<String> = Option::deserialize(deserializer)?;
+        match string {
+            Some(s) => {
+                let deserializer = serde::de::value::StringDeserializer::new(s);
+                uri_serde::deserialize(deserializer).map(Some)
+            },
+            None => Ok(None),
+        }
+    }
+}
```
Comment (lpil) [inline compiler-core/src/config.rs:841]: How come a custom decoder is needed here?

File: test/project_deno/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  - in src
+  - in test
+- Importing Gleam src code into test
+
+## Quick start
+
+```sh
+gleam run
+gleam test
+```
```
Comment (lpil) [inline test/project_deno/README.md:32]: This README file doesn't seem to be accurate

File: test/project_deno/gleam.toml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name = "project"
+version = "0.1.0"
+target = "javascript"
+
+[javascript]
+typescript_declarations = true
+runtime = "deno"
+
+[javascript.deno]
+location = "http://localhost:8080"
```
Comment (lpil) [inline test/project_deno/gleam.toml:10]: This doesn't seem to get used anywhere!

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+### Features
+- Update Deno config to allow passing `--location` runtime flag.
```
Comment (lpil) [inline CHANGELOG.md:6]: Could you put this under "build tool" please. Changes are categorised by what area of the codebase they are related to.

**Review by lpil — COMMENTED** [review]
Thank you!

The Deno test project doesn't test anything at the moment- is it unfinished? I would be happy to not have it or for it to test the Deno configuration specifically.

File: compiler-core/src/config.rs (hunk 2 above)
Comment (joshgillies) [reply compiler-core/src/config.rs:841]: Mostly due to inexperience. Rust isn't my strong suite, and admittedly this was the path of least resistance for getting tests to pass. I can spend a bit more time on it this evening and attempt to merge decoding `Option<Uri>` into the `uir_serde` decoder if that's preferable.

File: CHANGELOG.md (hunk 5 above)
Comment (joshgillies) [reply CHANGELOG.md:6]: Good call. Will update now 👍🏻

**Comment by joshgillies** [comment]
@lpil:
> The Deno test project doesn't test anything at the moment- is it unfinished?

Admittedly it's currently just a `cp -R` from the other `project_javascript` folder, and really only serves as a test that the `--location` flag is correctly passed to the `deno run` command via `gleam run` and `gleam test`.

> I would be happy to not have it or for it to test the Deno configuration specifically.

I'd be happy to remove it, and focus energy on a more robust testing strategy for Deno config in anther PR if that's an option. Otherwise I'd be happy to update the `project_deno` to concretely test the availability of the `location` API as documented [here](https://docs.deno.com/runtime/manual/runtime/location_api).

File: test/project_deno/gleam.toml (hunk 4 above)
Comment (joshgillies) [reply test/project_deno/gleam.toml:10]: Just pushed some tests to assert location API is defined correctly via this config option.

File: test/project_deno/README.md (hunk 3 above)
Comment (joshgillies) [reply test/project_deno/README.md:32]: Not 100% sure what we'd want to cover here, maybe something along the lines of:

> This project is to test various Deno runtime configuration settings.

File: compiler-core/src/config.rs (hunk 2 above)
Comment (joshgillies) [reply compiler-core/src/config.rs:835]: Have just removed the `option_uri_serde` decoder, instead merging the `Option<Uri>` logic into the base `uri_serde` decoder. Thanks for the nudge to clean that up. I wasn't totally happy making a new decoder for this case 😅

**Comment by joshgillies** [comment]
@lpil - have now updated the README. Please let me know if there's anything else you'd like done here.

File: test/project_deno/README.md (hunk 3 above)
Comment (joshgillies) [reply test/project_deno/README.md:15]: Readme has now been updated 🎉

**Review by lpil — APPROVED** [review]
Thank you !!

**Pull request merged**